feat: add register context and improve selection handling

deathbeam opened the pull request:
Adds new register context that allows accessing vim register contents in chat
and improves selection handling by:

- Making selection fields required instead of optional
- Improving unnamed register selection to include buffer context
- Deprecating clipboard selection in favor of register context
- Adding proper mark setting for better selection tracking
- Supporting array of default contexts in config

The register context supports all standard vim registers and provides a UI
for selecting them when used. This makes accessing clipboard and other
registers more intuitive.

jellydn reviewed (approved):
LGTM!

@deathbeam Could you update this usage on README as well?

deathbeam commented:
> LGTM!
> 
> @deathbeam Could you update this usage on README as well?

updated API section a bit but not sure what else needs updating, i added it to README to contexts and rest of context stuff is already explained there I think

deathbeam commented:
Ok swapped unnamed register back to selection as we can get position info for it via `[` and `]`
